### PR TITLE
searching specifications

### PR DESCRIPTION
**Pull request by BolsunovaNataliia**

### CODE REVIEW

File: src/main/java/mate/academy/bookshop/repository/book/spec/AuthorSpecificationProvider.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import mate.academy.bookshop.model.Book;
+import mate.academy.bookshop.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "author";
```
Comment (kshuryhin) [inline src/main/java/mate/academy/bookshop/repository/book/spec/AuthorSpecificationProvider.java:15]: good candidate for being constant

File: src/main/java/mate/academy/bookshop/repository/book/spec/PriceSpecificationProvider.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import mate.academy.bookshop.model.Book;
+import mate.academy.bookshop.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PriceSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "price";
```
Comment (kshuryhin) [inline src/main/java/mate/academy/bookshop/repository/book/spec/PriceSpecificationProvider.java:15]: good candidate for being constant

File: src/main/java/mate/academy/bookshop/repository/book/spec/TitleSpecificationProvider.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import mate.academy.bookshop.model.Book;
+import mate.academy.bookshop.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TitleSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "title";
```
Comment (kshuryhin) [inline src/main/java/mate/academy/bookshop/repository/book/spec/TitleSpecificationProvider.java:15]: good candidate for being constant

**Review by kshuryhin — CHANGES_REQUESTED** [review]
good job, just some minor points to consider

**Review by d1sam — APPROVED** [review]
Well done!

**Pull request merged**